docs/DOCUMENTACION.md

```diff
@@ -14,15 +14,29 @@ A través de la web de la Junta de Andalucía podremos obtener los datos relativ
\ No newline at end of file
+En el primer milestone(M0) se pretende realizar una representación del modelo del problema. En este caso se creará la implementación de una clase llamada tramo y otra clase ruta, siendo una ruta una colección de tramos entre un origen y un destino. Una vez obtenida la representación del modelo, en los milestones M1,M2,M3 se creará un grafo de tramos que conecten el lugar de origen con el destino y se aplicará la lógica de negocio mediante un algoritmo greedy para obtener las rutas descritas en las historias de usuario. En el milestone M1 se buscará la ruta más barata, en el milestone M2 se buscará la ruta más corta y en el milestone M3 la ruta más óptima. En cada uno de estos milestones se devolverá al usuario la ruta generada por pantalla.
+
+## Gestor de Dependencias
+A la hora de la elegir el gestor de dependencias para la aplicación es necesario establecer los criterios que que se van a tener en cuenta para esta eleción. En este caso se van a buscar gestores de dependencias que nos proporcionen una alta velocidad a la hora de las instalaciones y que no nos generen ficheros innecesarios en la aplicación. Teniendo estos criterios en cuenta, los gestores de dependencias que más se ajustan a lo anterior son los siguientes:
+- NPM: Gestor de dependencias por defecto de NodeJS, no nos genera deuda técnica debido a que ya viene instalado aunque ofrece un rendimiento muy bajo con respecto a pnpm y yarn.
+- PNPM: Las dependencias se almacenan en un almacén direccionable por contenido, por lo que a la hora de actualizar una dependencia se actualizará únicamente esa dependencia en lugar de clonar todas las
+dependencias para únicamente guardar ese cambio. Esto produce un ahorro de espacio en disco y una mejora sustancial del rendimiento. Su instalación es muy sencilla y no nos incluye nuevos ficheros en la aplicación.
+- Yarn: Es un gestor de paquetes(dependencias) que te permite usar y compartir código para aumentar un poco el rendimiento, además permite instalar paquetes en paralelo a diferencia de NPM que los instala de forma secuencial por lo que aumenta el rendimiento. Actualmente su uso está creciendo bastante y su instalación no requiere de la incorporación de uevos ficheros en la aplicación.
+
+Teniendo todo lo anterior en cuenta, para la aplicación se va a usar yarn debido a que su instalación no genera deuda técnica y nos da una mejora sustancial de rendimiento y seguridad con respecto a NPM, aunque con respecto a NPMP el rendimiento es similar.
+
+## Gestor de Tareas
+Tras elegir el gestor de Dependencias, es hora de elegir el Task Runner para la aplicación, para no aumentar la deuda técnica añadiendo nuevos ficheros a la aplicación, voy a usar yarn como Task Runner ya que para lo que se va a realizar en la asignatura es suficiente. Otras opciones sería usar Grunt pero nos encontramos el inconveniente de que tendríamos que incorporar el archivo Gruntfile.js.
```

Comment:
¿`D`ependencias?
Se trata en todo caso, como se ha repetido muchas veces, que pongáis los criterios y mostréis que conocéis el panorama y las opciones posibles.